wagtail_newsletter/wagtail_hooks.py

```diff
@@ -67,3 +67,10 @@ def redirect_to_campaign_page(request, page: Page):
 
     if action == "send_campaign":
         actions.send_campaign(request, page)
+
+
+@hooks.register("after_copy_page")  # type: ignore
+def clear_campaign_after_copy(request, page, new_page):
+    if isinstance(new_page, NewsletterPageMixin) and new_page.newsletter_campaign:
+        new_page.newsletter_campaign = ""
```

Comment:
It's only now that I notice that `newsletter_campaign` is a string - I reckon it's a unique identifier for the Campaign on the newsletter service's side (e.g. Mailchimp) ?  
(given that its max length is 1.000 I'm not sure to understand that right, though :sweat_smile: )

Maybe a `help_text` or a code comment on that model's field could help future developers working on this to understand the nature of that field?

Reply:
Good catch. I've added a comment on the model.

Perhaps 1000 is overkill, but it's certainly safe 😅

Reply:
Ah, so it was well and truly a "foreign" ID! :slightly_smiling_face:  

While we're here, as this is not super obvious I wonder if that field couldn't have a more explicit name to avoid potential future confusions - as `campaign` would be the name of a linked Campaign model (via an underlying `campaign_id` field) in Django's conventions?

Something like `campaign_external_id`, or something like this?  
What do you think?

Reply:
I've thought about this some too. To keep the naming convention that all fields defined by `NewsletterPageMixin` begin with `newsletter_`, it would have to be `newsletter_campaign_external_id`, which feels kind of long. I'm not so sure that `newsletter_campaign` is _that_ confusing, TBH.

Reply:
Fair enough! :slightly_smiling_face: